modules/ui/src/main/scala/lucuma/ui/components/SideTabs.scala

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.ui.components
+
+import cats.syntax.all.*
+import crystal.react.hooks.*
+import japgolly.scalajs.react.*
+import japgolly.scalajs.react.vdom.html_<^.*
+import lucuma.core.util.Display
+import lucuma.core.util.Enumerated
+import lucuma.react.common.Css
+import lucuma.react.common.ReactFnProps
+import lucuma.refined.*
+import lucuma.ui.primereact.*
+import lucuma.ui.primereact.given
+import lucuma.ui.syntax.all.given
+
+/**
+ * Component to display side tabs based on an enumerated (Order is important) Each tab can be
+ * selected and will be highlighted and it will push a page possibly via Routing
+ *
+ * The `separatorAfter` function is used to determine if a separator should be displayed after the
+ */
+case class SideTabs[A](
+  appTab:         A,
+  pushPage:       A => Callback,
+  pageUrl:        A => String,
+  separatorAfter: A => Boolean
+)(using val enumerated: Enumerated[A], val display: Display[A])
+    extends ReactFnProps(SideTabs.component)
+
+object SideTabs:
+  private type AnyF[_] = Any
+
+  private type Props[A] = SideTabs[A]
+
+  private def buildComponent[V[_], A] =
+    ScalaFnComponent
+      .withHooks[Props[A]]
+      .useStateViewBy(_.enumerated.all.head)
+      .useEffectWithDepsBy((p, _) => p.enumerated.tag(p.appTab))((p, selected) =>
+        focus => selected.set(p.enumerated.unsafeFromTag(focus))
+      )
+      .render { (p, tabs) =>
+        import p.given
+
+        val tabView = tabs.withOnMod(p.pushPage)
+
+        <.div(
+          SideTabsStyles.SideTabs,
+          <.div(
+            SideTabsStyles.SideTabsVertical,
+            SelectButtonEnumView(
+              "side-tabs".refined,
```

Comment:
I can't forsee having more than one side-tabs group at the same time, but just in case, should the `id` be a property?